master-runner/src/main/scala/de/thm/ii/fbs/services/runner/BashRunnerService.scala

```diff
@@ -0,0 +1,116 @@
+package de.thm.ii.fbs.services.runner
+
+import java.nio.file.Path
+
+import de.thm.ii.fbs.services.{DockerService, FileService}
+import de.thm.ii.fbs.types.{DockerCmdConfig, Runner, Submission}
+import de.thm.ii.fbs.util.RunnerException
+
+/**
+  * BashRunnerService provides all functions to start a Bash Runner
+  *
+  * @param runner     Runner Configuration
+  * @param submission the Submission to Check
+  * @author Max Stephan
+  */
+class BashRunnerService(val runner: Runner, val submission: Submission) {
+  private val MAIN_FILE_MOUNT = s"/runner/main-file"
+  private val SECONDARY_FILE_MOUNT = s"/runner/secondary-file"
+  private val SOURCE_FOLDER = s"/runner"
+  private var tmpFolder: Option[Path] = None
+
+  /**
+    * Create a DockerCmd Configuration for a BashRunner
+    *
+    * @return DockerCmd Config
+    */
+  def getDockerCmd: DockerCmdConfig = {
+    /*Get Docker Options*/
+    // Read the first line of the Script File to check if is a php script TODO May improve/remove
+    val scriptContent = FileService.fileToString(submission.solutionFileLocation.toFile)
+    val interpreter = if (scriptContent.split("\n").head.matches("#!.*php.*")) "php" else "bash"
+
+    val infoArgument = if (submission.isInfo) "info" else ""
+    val submissionMount = s"$SOURCE_FOLDER/${submission.solutionFileLocation.getFileName}"
+
+    /* Build Docker Command */
+    var mountFiles = Seq(
+      (runner.mainFile.toString, MAIN_FILE_MOUNT),
+      (submission.solutionFileLocation.toString, submissionMount))
+
+    if (runner.hasSecondaryFile) {
+      mountFiles = mountFiles :+ (runner.secondaryFile.toString, SECONDARY_FILE_MOUNT)
+    }
+
+    // If secondary file is present add its path to an environment Variable
+    val dockerOptions = if (runner.hasSecondaryFile) {
+      Seq("--env", s"TESTFILE_PATH=$SECONDARY_FILE_MOUNT")
+    } else {
+      Seq.empty
+    }
+
+    val runOptions = Seq(interpreter, MAIN_FILE_MOUNT, submission.user.username, submissionMount, infoArgument)
```

Comment:
- i have not changed the argument order/structure of the bash file (so all scripts are compatible). However, I think it is a bit inconsistent, should we change it? (currently it is: `username, submissionPath, infoArgument` and the Secondary file path is stored in the env `TESTFILE_PATH`.  Wouldn't it be better to pass the path of the second file as well?)